Tests/SonarScanner.MSBuild.PreProcessor.Test/Roslyn/PluginTests.cs

```diff
@@ -29,66 +28,64 @@ namespace SonarScanner.MSBuild.PreProcessor.Test.Roslyn;
 [TestClass]
 public class PluginTests
 {
-    [TestMethod]
-    public void Ctor_WhenKeyIsInvalid_Throws()
-    {
-        // 1. Null
-        Action act = () => new Plugin(null, "v1", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
-
-        // 2. Empty
-        act = () => new Plugin(string.Empty, "v1", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
-
-        // 3. Whitespace
-        act = () => new Plugin("\r ", "v1", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
-    }
-
-    [TestMethod]
-    public void Ctor_WhenVersionIsInvalid_Throws()
+    [DataTestMethod]
+    [DataRow("pluginKey", "42", "test.zip", true)]
+    [DataRow("pluginKey", "42", null, false)]
+    [DataRow("pluginKey", null, "test.zip", false)]
+    [DataRow(null, "42", "test.zip", false)]
+    [DataRow("pluginKey", null, null, false)]
+    [DataRow(null, "42", null, false)]
+    [DataRow(null, null, "test.zip", false)]
+    [DataRow(null, null, null, false)]
+    public void Plugin_IsValid(string key, string version, string resourceName, bool isValid)
     {
-        // 1. Null
-        Action act = () => new Plugin("key", null, "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("version");
-
-        // 2. Empty
-        act = () => new Plugin("key", string.Empty, "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("version");
-
-        // 3. Whitespace
-        act = () => new Plugin("key", "\r\n ", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("version");
+        var plugin = new Plugin() { Key = key, Version = version, StaticResourceName = resourceName };
+        plugin.IsValid.Should().Be(isValid);
     }
 
-    [TestMethod]
-    public void Ctor_WhenStaticResourceNameIsNull_Throws()
+    [DataTestMethod]
+    [DataRow("pluginKey", "someValue")]
+    [DataRow("pluginVersion", "42.0.0")]
+    [DataRow("staticResourceName", "test.zip")]
+    [DataRow("someOtherProperty", "someOtherValue")]
+    public void AddProperty_Populates_Correctly(string property, string value)
```

Comment:
This has a lot of straighteners applied on benders. Just split it in 4 UTs

Reply:
Or actually, just call the AddProperty 4 times, and then assert the state of the object. That will be much easier. We don't need the `BeNull()` parts